src/auro-select.js

```diff
@@ -153,12 +153,23 @@ class AuroSelect extends LitElement {
     });
   }
 
-  updated() {
+  updated(changedProperties) {
+    if (changedProperties.has('value')) {
+      let updateValueMatch = false;
+
+      if (this.value) {
+        if (!this.optionSelected) {
+          // If there is a value but no option selected, make the selection.
+          updateValueMatch = true;
+        } else if (this.value !== this.optionSelected.value) {
+          // If there is a value and a selection but they don't match, update the selection to match the value.
+          updateValueMatch = true;
+        }
+      }
 
-    /** Check for preset value attribute.
-     * Have auro-menu attempt a selection of that value */
-    if (this.value !== this.displayValue) {
-      this.menu.selectByValue(this.value);
+      if (updateValueMatch) {
+        this.menu.selectByValue(this.value);
+      }
```

Comment:
The way this is written, it took a while for me to realize there's only one case where we run the function -- the multiple levels of `if` nesting threw me off. Could we do something like this instead?

```js
if (this.value && (!this.optionSelected || this.value !== this.openSelected.value)) {
  this.menu.selectByValue(this.value);
}
```

Also, this component is getting complex enough that we _really_ need some tests (#38). Would it be possible to at least add a test for what was added in this PR?

Reply:
I felt breaking it down with comments was more clear but I don't mind doing it the other way either. I do appreciate how it's more succinct this way.

Regarding the test, we have a hurdle with both select and combobox we have to sort out regarding testing with the child components. It's a high priority task to get both components up on their tests but we have to solve the issue we are having to get the tests to run right. They will be updated for both components very soon.